benchmark/cmd/benchmark.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+
+	"k8s.io/perf-tests/benchmark/pkg/comparer"
+	"k8s.io/perf-tests/benchmark/pkg/runselector"
+	"k8s.io/perf-tests/benchmark/pkg/scraper"
+	"k8s.io/perf-tests/benchmark/pkg/util"
+
+	"github.com/golang/glog"
+	"github.com/spf13/pflag"
+)
+
+var (
+	leftJobName, rightJobName string
+	runSelectionScheme        string
+	nValue                    int
+	comparisonScheme          string
+	matchThreshold            float64
+)
+
+func registerFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&leftJobName, "left-job-name", "ci-kubernetes-e2e-gce-scalability", "Name of the job to be used as left hand side of comparison")
+	fs.StringVar(&rightJobName, "right-job-name", "ci-kubernetes-kubemark-100-gce", "Name of the job to be used as right hand side of comparison")
+	fs.StringVar(&runSelectionScheme, "run-selection-scheme", runselector.LastNHours, "Scheme for selecting the runs to be compared")
+	fs.IntVar(&nValue, "n", 24, "Value of 'n' (runs/hours depending on run-selection-scheme) to be used for run selection")
```

Comment:
It shows that the design is wrong. You should just have two flags: `--last-runs-number` and `--last-runs-hours`, which are mutually exclusive.